lib/Request/CachingClient.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace DCG\Cinema\Request;
+
+use DCG\Cinema\Cache\CacheInterface;
+use DCG\Cinema\Request\Cache\KeyGenerator;
+use DCG\Cinema\Request\Cache\LifetimeGenerator;
+
+class CachingClient implements ClientInterface
+{
+    const CACHE_PREFIX = 'cinema:request:';
+
+    private $client;
+    private $cache;
+    private $keyGenerator;
+    private $lifetimeGenerator;
+
+    public function __construct(
+        ClientInterface $client,
+        CacheInterface $cache,
+        KeyGenerator $keyGenerator,
+        LifetimeGenerator $lifetimeGenerator
+    ) {
+        $this->client = $client;
+        $this->cache = $cache;
+        $this->keyGenerator = $keyGenerator;
+        $this->lifetimeGenerator = $lifetimeGenerator;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function get($path, $queryParams = [], $successStatusCodes = [200])
+    {
+        $cacheKey = $this->keyGenerator->generateKey('get', $path, $queryParams, $successStatusCodes);
+        $result = $this->cache->get($cacheKey);
```

Comment:
Is null a valid thing to store in cache? If so, you should check if key exists as well as getting